src/libraries/Microsoft.PowerFx.Core/Public/DisplayNames/DisplayNameProvider.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.PowerFx.Core.UtilityDataStructures;
+
+namespace Microsoft.PowerFx.Core
+{
+    internal class DisplayNameProvider
+    {
+        // String comparison is case-sensitive by default, and that's the behavior we want
+        private BidirectionalDictionary<string, string> _displayNames;
```

Comment:
>string [](http://example.com/codeflow?start=40&length=6)

nit:
- comment what the key & value are
- is the comparison case-sensitive or not?

Reply:
added constructor with explicit comparator, and comments